spec/unit/http/session_spec.rb

```diff
@@ -83,4 +83,49 @@ def check_connection?(session, service, ssl_context: nil)
       }.to raise_error(ArgumentError, "Unknown service westbound")
     end
   end
+
+  context 'when resolving using multiple resolvers' do
+    let(:session) { client.create_session }
+
+    it "prefers SRV records" do
+      Puppet[:use_srv_records] = true
+
+      allow_any_instance_of(Puppet::Network::Resolver).to receive(:each_srv_record).and_yield('mars.example.srv', 8140)
+      service = session.route_to(:ca)
+
+      expect(service.url).to eq(URI("https://mars.example.srv:8140/puppet-ca/v1"))
+    end
+
+    it "next prefers the first successful connection from server_list" do
+      Puppet[:use_srv_records] = true
+      Puppet[:srv_domain] = "srv.example.com"
+      Puppet[:server_list] = 'foo.example.com,bar.example.com,baz.example.com'
+
+      stub_request(:get, "https://foo.example.com:8140/status/v1/simple/master").to_return(status: 500)
+      stub_request(:get, "https://bar.example.com:8140/status/v1/simple/master").to_return(status: 200)
+
+      service = session.route_to(:ca)
+
+      expect(service.url).to eq(URI("https://bar.example.com:8140/puppet-ca/v1"))
+    end
+
+    it "next prefers settings" do
+      Puppet[:use_srv_records] = true
+      Puppet[:srv_domain] = "srv.example.com"
+      Puppet[:ca_server] = 'caserver.example.com'
+
+      service = session.route_to(:ca)
+
+      expect(service.url).to eq(URI("https://caserver.example.com:8140/puppet-ca/v1"))
+    end
+
+
+    it "raises when there are no more routes" do
```

Comment:
I'm not sure we're ever going to raise here. It looks like `check_connection?` will always return true for us in these tests, which is throwing me off. Because of that, `puppet:8140` will get returned as a valid route